### PR TITLE
[11.x] Add trustedproxy config file to allow dynamic assignment for IPs

### PR DESCRIPTION
**Pull request by omaresmael**
This pr adds `trustedproxy.php` config file to Laravel config files so it can be published with `php artisan config:puplish`

### TL;DR
Laravel already depends on this config file to assign proxies, but the config file is missing and need to be created manually, the pr adds this config file. 

### The issue
I wanted to have a variable list of IPs per environment (dev, staging, production). The issue is you can not use either `env` nor `config` in the `boostrap/app.php` 
```php
Application::configure(basePath: dirname(__DIR__))
    ->withRouting(
        web: __DIR__.'/../routes/web.php',
        commands: __DIR__.'/../routes/console.php',
        health: '/up',
    )
    ->withMiddleware(function (Middleware $middleware) {
        $middleware->trustProxies(env("TRUSTED_IPS"));  // env will return null
        $middleware->trustProxies(config("trustedproxy.proxies"));  // config will throw ReflectionException 
    })
    ->withExceptions(function (Exceptions $exceptions) {
        //
    })->create();
```

I looked into the implementation of the `TrustProxies.php` middleware and found this interesting code

```php
 protected function setTrustedProxyIpAddresses(Request $request)
    {
        $trustedIps = $this->proxies() ?: config('trustedproxy.proxies'); // Laravel already supports getting ips from the trustedproxy.php config file

        if ($trustedIps === '*' || $trustedIps === '**') {
            return $this->setTrustedProxyIpAddressesToTheCallingIp($request);
        }

        $trustedIps = is_string($trustedIps)
                ? array_map('trim', explode(',', $trustedIps))
                : $trustedIps;

        if (is_array($trustedIps)) {
            return $this->setTrustedProxyIpAddressesToSpecificIps($request, $trustedIps);
        }
    }
```
### The Solution
I manually created the `trustedproxy.php` config file and passed to it the IPs per env.

### Suggestion

I think it will be more convenient to publish the config file if needed, instead of creating it manually. Hence, this pr. 

Let me know please if that is a good solution or If I'm missing something

Thanks ❤️

### CODE REVIEW

**Comment by taylorotwell** [comment]
You should be able to create this for your own project.

**Comment by illambo** [comment]
Duplicate of PR #52064.
